src/training/data.py

```diff
@@ -359,7 +361,8 @@ def get_wds_dataset(args, preprocess_img, is_train, epoch=0, floor=False):
         num_samples = num_batches * global_batch_size
         dataset = dataset.with_epoch(num_worker_batches)  # each worker is iterating over this
     else:
-        # last batches are partial, eval is done on single (master) node
+        if args.distributed_evaluation:
+            num_samples = num_samples // args.world_size
```

Comment:
maybe I'm misunderstanding, but does it skip the last few samples due to the last partial batch?

Reply:
> maybe I'm misunderstanding, but does it skip the last few samples due to the last partial batch?

in evaluation, `num_samples` is only used on logging (https://github.com/mlfoundations/open_clip/blob/03839c5e00b1c21c14dbaa07921d7190a28125c5/src/training/train.py#L172, https://github.com/mlfoundations/open_clip/blob/03839c5e00b1c21c14dbaa07921d7190a28125c5/src/training/train.py#L217), it is not affecting the dataloader which depend on the wds pipeline. But it would be good to get it correct anyway, have to check how many examples  each worker exactly receive.